### PR TITLE
basic icontrolLX plugin for interacting with bigip

### PR DESCRIPTION
**Pull request by elv1s**

### CODE REVIEW

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const url = require('url');
+const logger = require('f5-logger').getInstance();
+
+/**
+ * A simple REST Worker that demonstrates updating config on a bigip
+ * @constructor
+ */
+class HelloBigipWorker {
+    constructor() {
+        this.WORKER_URI_PATH = "Nodes";
```
Comment (manjubhat) [inline icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js:27]: use the URI helloBigIP ?

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    /**
+     * handle onGet HTTP request
+     * @param {Object} restOperation
+     */
+    onGet(restOperation) {
+        logger.info('/Nodes GET request');
+
+        this.getNodes(restOperation)
+        .then((nodes)=> {
```
Comment (manjubhat) [inline icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: rename nodes to response ? response.getBody() makes more sense for example

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        if (!body) {
+            throw new Error("no POST body");
+        }
+
+        if (!body.start || !body.end) {
+            throw new Error("body needs 'start' and 'end' values");
+        }
+
+        if (!(this.isValidIPv4(body.start) && this.isValidIPv4(body.end))) {
+            throw new Error("start and end are invalid IPv4 addresses");
```
Comment (manjubhat) [inline icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: nit: either of start or end are invalid IPv4 address. Need not be both

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+     *
+     * @param {Object} node
+     * @param {RestOperation} originalRestOp
+     * @returns {Promise}
+     */
+    createNode(node, originalRestOp) {
+        const nodeUrl = this.restHelper.makeRestjavadUri('tm/ltm/node');
+
+        const restOp = this.restOperationFactory.createRestOperationInstance()
+            .setUri(url.parse(nodeUrl))
```
Comment (manjubhat) [inline icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: this url.parse is unnecessary right ?

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+        return this.restRequestSender.sendPost(restOp);
+    }
+
+
+    getNodes(originalRestOp) {
+        const nodeUrl = this.restHelper.makeRestjavadUri('tm/ltm/node');
+
+        const restOp = this.restOperationFactory.createRestOperationInstance()
+            .setUri(url.parse(nodeUrl))
```
Comment (manjubhat) [inline icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: same as above

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js (hunk 1 above)
Comment (elv1s) [reply icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js:27]: I was thinking that, but Nodes is more descriptive for that the endpoint is doing.

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js (hunk 2 above)
Comment (elv1s) [reply icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: makes sense, fixed

File: icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js (hunk 3 above)
Comment (elv1s) [reply icontrollx/hello-bigip/src/nodejs/HelloBigipWorker.js]: fixed

**Review by manjubhat — COMMENTED** [review]
Shipit

**Pull request merged**